packages/syft/src/syft/core/smpc/store/__init__.py

```diff
@@ -0,0 +1,67 @@
+"""Package where needed crypto material is stored."""
+
+# stdlib
+# stdlib
+from typing import Any
+from typing import Callable
+
+# relative
+from .crypto_primitive_provider import CryptoPrimitiveProvider
+from .crypto_store import CryptoStore
+
+
+def register_primitive_generator(name: str) -> Callable[..., Any]:
+    """Decorator to register a crypto primitive provider.
+
+    Args:
+        name (str): Name of the primitive.
+
+    # noqa: DAR201
+    """
+
+    def register_generator(func_generator: Callable[..., Any]) -> Callable[..., Any]:
+        if name in CryptoPrimitiveProvider._func_providers:
+            raise ValueError(f"Provider {name} already in _func_providers")
+        CryptoPrimitiveProvider._func_providers[name] = func_generator
+        return func_generator
+
+    return register_generator
+
+
+def register_primitive_store_add(name: str) -> Callable[..., Any]:
+    """Decorator to add primitives to the store.
+
+    Args:
+        name (str): Name of the primitive.
+
+    # noqa: DAR201
```

Comment:
MAYBE: Could you add here the return type?

Reply:
Sure :+1: